compiler/src/test/scala/org/scalajs/core/compiler/test/OptimizationTest.scala

```diff
@@ -172,4 +172,55 @@ class OptimizationTest extends JSASTTest {
     }
   }
 
+  @Test
+  def newSJSDefinedTraitProducesObjectConstr: Unit = {
+    """
+    import scala.scalajs.js
+    import scala.scalajs.js.annotation._
+
+    @ScalaJSDefined
+    trait Point extends js.Object {
+      val x: Double
+      val y: Double
+    }
+
+    class Test {
+      def newSJSDefinedTraitProducesObjectConstr(): Any = {
+        new Point {
+          val x = 5.0
+          val y = 6.5
+        }
+      }
+    }
+    """.hasNot("`new Object`") {
+      case js.JSNew(_, _) =>
+    }.has("object literal") {
+      case js.JSObjectConstr(Nil) =>
```

Comment:
Are you sure you want to assert the `Nil` inside? Having the actual fields would be ok, too, wouldn't it?